src/components/molecules/LocationMap.tsx

```diff
@@ -1,32 +1,72 @@
 import 'leaflet-css'
 import React, { FunctionComponent } from 'react'
 import L from 'leaflet'
-import { Map, Marker, TileLayer } from 'react-leaflet'
+import { Map, TileLayer } from 'react-leaflet'
+import AnywayMarker from '../atoms/AnywayMarker'
 
 // TODO: Move to init.service once it's merged.
 L.Icon.Default.mergeOptions({
     iconRetinaUrl: require('leaflet/dist/images/marker-icon-2x.png'),
     iconUrl: require('leaflet/dist/images/marker-icon.png'),
     shadowUrl: require('leaflet/dist/images/marker-shadow.png')
 })
+const lIconSizes = { iconSize: [25, 41], iconAnchor: [12, 41], popupAnchor: [1, -34], shadowSize: [41, 41] }
+const iconSize = lIconSizes;
+
 
 interface IProps {
+    data: any[],
     marker: { lat: number; lng: number }
 }
 
 const INITIAL_ZOOM = 13
 const WRAPPER_STYLES = { height: '100%', width: '100%' }
 
-const LocationMap: FunctionComponent<IProps> = ({ marker }) => {
+const LocationMap: FunctionComponent<IProps> = ({ data, marker }) => {
+    let markers = data.map((x: any) => {
+        if (x.latitude !== null && x.longitude !== null) {
+            return <AnywayMarker markerdata={x} />
+        }
+        else return null;
+    });
+    const bounds = setBounds(data)
+
     return (
-        <Map center={marker} zoom={INITIAL_ZOOM} style={WRAPPER_STYLES}>
+        <Map center={marker} bounds={bounds} zoom={INITIAL_ZOOM} style={WRAPPER_STYLES}>
             <TileLayer
                 url='https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png'
                 attribution='&copy; <a href="http://osm.org/copyright">OpenStreetMap</a> contributors'
             />
-            <Marker position={marker} />
+            {markers}
         </Map>
     )
 }
-
+const setBounds = (data: any[]) => {
+    var bounds;
+    var corner1 = L.latLng(29.50, 34.22)
+    var corner2 = L.latLng(33.271, 35.946)
+    var arr:L.LatLng[] = [];
+    var lastPoint:L.LatLng = L.latLng(0,0) ;
+    data.forEach(x => {
```

Comment:
This feels like something that should be done with `reduce`.

Reply:
we need lastPoint in case we get single point (this can be even is array has some points  - they can have same cordinates).
in all other cases we need an array of points.

Reply:
You can access the last point with arr[arr.length -1].

Reply:
in the begining i  used arr.pop()  but it gave becouse it van be null, and we also need lastpoint to prevnent duplact,